nixos/modules/services/mail/listmonk.nix

```diff
@@ -0,0 +1,212 @@
+              type = listOf str;
+              default = [];
+              description = lib.mdDoc "E-mail addresses with these domains are disallowed from subscribing.";
+            };
+
+            smtp = mkOption {
+              type = listOf (submodule {
+                freeformType = with types; attrsOf (oneOf [ str int bool ]);
+
+                options = {
+                  enabled = mkEnableOption "this SMTP server for listmonk";
+                  host = mkOption {
+                    type = types.str;
+                    description = lib.mdDoc "Hostname for the SMTP server";
+                  };
+                  port = mkOption {
+                    type = types.port;
+                    description = lib.mdDoc "Port for the SMTP server";
+                  };
+                  max_conns = mkOption {
+                    type = types.int;
+                    description = lib.mdDoc "Maximum number of simultaneous connections, defaults to 1";
+                    default = 1;
+                  };
+                  tls_type = mkOption {
+                    type = types.enum [ "none" "STARTTLS" "TLS" ];
+                    description = lib.mdDoc "Type of TLS authentication with the SMTP server";
+                  };
+                };
+              });
+
+              description = lib.mdDoc "List of outgoing SMTP servers";
+            };
+
+            # TODO: refine this type based on the smtp one.
+            "bounce.mailboxes" = mkOption {
+              type = listOf (submodule {
+                freeformType = with types; oneOf [ str int bool ];
+              });
+              default = [];
+              description = lib.mdDoc "List of bounce mailboxes";
+            };
+
+            messengers = mkOption {
+              type = listOf str;
+              default = [];
+              description = lib.mdDoc "List of messengers, see: <https://github.com/knadh/listmonk/blob/master/models/settings.go#L64-L74> for options.";
+            };
+          };
+        });
+
+        description = lib.mdDoc "Dynamic settings in the PostgreSQL database, set by a SQL script, see <https://github.com/knadh/listmonk/blob/master/schema.sql#L177-L230> for details.";
+      };
+      secretFile = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = lib.mdDoc "Secret file loaded as an environment file, read <https://listmonk.app/docs/configuration/#environment-variables> for details.";
+      };
+    };
+  };
+
+
+  ###### implementation
+  config = mkIf cfg.enable (mkMerge [
+    {
+      services.listmonk.settings."app".address = mkDefault "localhost:9000";
+
+      # Default parameters from https://github.com/knadh/listmonk/blob/master/config.toml.sample
+      services.listmonk.settings = {
+        "db".max_open = mkDefault 25;
+        "db".max_idle = mkDefault 25;
+        "db".max_lifetime = mkDefault "300s";
+      };
+
+      users.users."listmonk" = {
```

Comment:
This doesn't make sense with `DynamicUser`. I understand what you're doing here... you shouldn't muck with user accounts this way. Aside from this likely causing issues down the road I wouldn't feel confident this is entirely secure without thinking it through.

Drop `DynamicUser` please.

Reply:
Sure thing.